src/components/EditButton.js

```diff
@@ -0,0 +1,42 @@
+import {
+  Image,
+  TouchableOpacity,
+  StyleSheet,
+  ViewPropTypes,
+  TouchableHighlight,
+} from 'react-native';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+function EditButton({ style, onPress }) {
+  return (
+    <TouchableOpacity
+      style={{ ...styles.edit_button_container, ...style }}
+      onPress={onPress}
+    >
+      <Image
+        source={require('../../assets/icons/edit.png')}
+        style={styles.edit_button}
+        resizeMode={'contain'}
+      />
+    </TouchableOpacity>
+  );
+}
+export default EditButton;
+
+EditButton.propTypes = {
+  style: TouchableHighlight.propTypes.style,
+  onPress: PropTypes.func.isRequired,
+};
+
```

Comment:
Let's keep it all similar
1. Component/Function
2. Styles
3. PropTypes

Reply:
Styles should be defined before they are used

Reply:
1. Imports
2. Const variables
3. Component/function definition
4. Export default <component/function name>
5. Prop types
6. Default props

Reply:
Why place styles above the component?
https://github.com/airbnb/javascript/tree/master/css-in-javascript  -> Ordering
I think is normal to go from the abstraction down to details.
This way you can read read just the top of the file to get general idea about what it is doing, and if you need to get deeper you just go down to details like functions/styles.
I don't  know js so correct my if I'm wrong

Reply:
I see your point and I think the concept you mentioned is widely used in object-oriented programming. In pure javascript we use functional programming. 

Styles should go above the component due to https://eslint.org/docs/rules/no-use-before-define.

In javascript we first declare sth and then we use it. This rule is applied to variables and functions. I think it make sens. I read styles then I read function that uses the styles and I know what are styles.